uvicorn/protocols/http/h11_impl.py

```diff
@@ -155,6 +155,21 @@ def _unset_keepalive_if_required(self) -> None:
             self.timeout_keep_alive_task.cancel()
             self.timeout_keep_alive_task = None
 
+    def _should_upgrade(self, event: H11Event) -> bool:
```

Comment:
Neither the event is used 🤔 

```suggestion
    def _should_upgrade(self) -> bool:
```

Reply:
Yes that is done in accordance to my proposal to enable easy overwriting for more complex use cases (s. #1659).
Would you prefer simplicity over customizability?

Reply:
It's a private method, I don't think we should think about customization here. 🤔 

But... Then this PR doesn't fit your needs? 🤔

Reply:
The PR will still fit my needs, however when thinking about the implementation I thought maybe someone will be happy because they only want to ignore on certain paths. During implementation I realized it won't be as simple for the httptools protocol.
YAGNI? :D

We could then think of extracting the method to a common place for both protocols, since they're identical

Reply:
In case you need to do something custom, uvicorn supports a way to implement your own HTTP protocol class:

```python
uvicorn.run("main:app", http=CustomHTTPProtocol)
```
jfyk

Reply:
> YAGNI

Yep, please. 👀

Reply:
Want me to keep the duplication or introduce a utils file (`upgrade.py`?)

Reply:
would make it necessary to pass the headers and config around again though

Reply:
keep it for now, before merge I can rearrange if we change our minds, dw 👍

Just removing the event is fine